forge/src/main/java/whocraft/tardis_refined/common/data/ProviderLootTable.java

```diff
@@ -20,31 +18,22 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
-import java.util.function.BiConsumer;
-import java.util.function.Consumer;
-import java.util.function.Supplier;
+import java.util.Set;
+import java.util.stream.Stream;
 
 public class ProviderLootTable extends LootTableProvider {
 
-    public ProviderLootTable(DataGenerator dataGenerator) {
-        super(dataGenerator);
+    public ProviderLootTable(PackOutput arg) {
+        super(arg, Set.of(), List.of(new LootTableProvider.SubProviderEntry(ModBlockLoot::new, LootContextParamSets.BLOCK), new LootTableProvider.SubProviderEntry(ModEntityLoot::new, LootContextParamSets.ENTITY)));
     }
 
-    @Override
-    protected List<Pair<Supplier<Consumer<BiConsumer<ResourceLocation, LootTable.Builder>>>, LootContextParamSet>> getTables() {
-        return List.of(Pair.of(ModEntityLoot::new, LootContextParamSets.ENTITY), Pair.of(ModBlockLoot::new, LootContextParamSets.BLOCK));
-    }
-
-    @Override
-    protected void validate(Map<ResourceLocation, LootTable> map, ValidationContext validationContext) {
-        for (Map.Entry<ResourceLocation, LootTable> entry : map.entrySet())
-            LootTables.validate(validationContext, entry.getKey(), entry.getValue());
-    }
+    public static class ModBlockLoot extends BlockLootSubProvider {
+        protected ModBlockLoot() {
+            super(Set.of(), FeatureFlags.REGISTRY.allFlags());
```

Comment:
Isn't the Set.of() meant to be a set of modids?

Reply:
That set is for explosion-resistant blocks. We could add the console and shells to it but I don't *think* that is necessary, I will check.